src/ModelMerger.py

```diff
@@ -0,0 +1,158 @@
+
+    def translate_ids(self, score_thr, score_type="Name_score"):
+        """
+        Translates metabolite IDs in self.cobra_model based on a score threshold and updates corresponding
+        translation reaction IDs.
+
+        Parameters:
+            score_thr (float): Minimum score required for ID translation.
+            score_type (str): Score type to filter matches by, must be a valid column in the matches table
+
+        Raises:
+            ValueError: If the specified score type is not found in the matches table or if a metabolite has
+                        more than one or no "TR_" reaction.
+        """
+        if score_type not in self.matches.columns: # makes sure that the chosen score is specified in the matches table
+            raise ValueError("Specified score type: " + score_type + "doesn't exist")
+        cobra_model = self.memo_model.cobra_model
+        reliable_matches = self.matches.loc[self.matches[score_type] >= score_thr]
+        #  selects the metabolites to be translated
+        to_translate = list()
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                to_translate = to_translate + [met]
+        # applies the translation to the metabolites
+        self.translate_metabolites(to_translate, reliable_matches, score_type)
+        #  translate the ids of the respective TR_ reactions
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                tr_rxns_for_current_met = [rxn for rxn in met.reactions if rxn.id.startswith("TR_")]
+                if len(tr_rxns_for_current_met) != 1:
+                    raise ValueError(met.id + " should have one and only one TR_ reaction, it has " +
+                                     str(len(tr_rxns_for_current_met)) + " instead")
+                tr_rxn = tr_rxns_for_current_met[0]
+                tr_rxn.id = "TR_" + met.id
+
+    def create_exchanges(self):
+        cobra_model = self.memo_model.cobra_model
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                reaction_id = "EX_" + met.id
+                rxn = cobra.Reaction(id=reaction_id, name=met.name + " exchange", lower_bound=-1000, upper_bound=1000)
+                rxn.add_metabolites({met: -1})
+                cobra_model.add_reactions([rxn])
+
+    def set_rxn_bounds(self):
+        cobra_model = self.memo_model.cobra_model
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                ex_rxn = cobra_model.reactions.get_by_id("EX_" + met.id)
+                tr_rxn = cobra_model.reactions.get_by_id("TR_" + met.id)
+                ex_rxn.lower_bound = tr_rxn.lower_bound
+                ex_rxn.upper_bound = tr_rxn.upper_bound
+                tr_rxn.lower_bound = -1000
+                tr_rxn.upper_bound = 1000
+
+
+    def translate_namespace(self):
+        self.convert_exchange_rxns_to_translation_rxns()
```

Comment:
Add an example what happens to a metabolite/reaction after each of those steps. Tell a little story of beautiful small metabolite on it journey through ModelMerging land

Reply:
I.e. ADd a comment after ther first function call that tells what happens to the ex reactions and why, and that we lose the actual ex reactions for now